src/deepsparse/transformers/pipelines/question_answering.py

```diff
@@ -233,80 +296,219 @@ def process_engine_outputs(
+        if len(predictions) == 0 or (
+            len(predictions) == 1 and predictions[0]["text"] == ""
+        ):
+            predictions.insert(
+                0, {"text": "empty", "start_logit": 0.0, "end_logit": 0.0, "score": 0.0}
+            )
+
+        # Compute the softmax of all scores (we do it with numpy to stay independent
+        # from torch/tf in this file, using the LogSumExp trick)
+        scores = numpy.array([pred.pop("score") for pred in predictions])
+        exp_scores = numpy.exp(scores - numpy.max(scores))
+        probs = exp_scores / exp_scores.sum()
 
-            char_start = _token_to_char(ans_start)
-            while char_start is None:
-                ans_start += 1
-                char_start = _token_to_char(ans_start)
-
-            char_end = _token_to_char(ans_end)
-            while char_end is None:
-                ans_end += 1
-                char_end = _token_to_char(ans_end)
-
-            return self.output_schema(
-                score=score.item(),
-                start=char_start[0],
-                end=char_end[1],
-                answer=example.context_text[char_start[0] : char_end[1]],
+        # Include the probabilities in our predictions.
+        for prob, pred in zip(probs, predictions):
+            pred["probability"] = prob
+
+        # The dictionaries we have to fill.
+        all_predictions = collections.OrderedDict()
+        all_nbest_json = collections.OrderedDict()
+
+        # Pick the best prediction. If the null answer is not possible, this is easy.
+        if not self.version_2_with_negative:
+            all_predictions[example.qas_id] = predictions[0]["text"]
+        else:
+            # Otherwise we first need to find the best non-empty prediction.
+            i = 0
+            while predictions[i]["text"] == "":
+                i += 1
+            best_non_null_pred = predictions[i]
+
+            # Then we compare to the null prediction using the threshold.
+            score_diff = (
+                null_score
+                - best_non_null_pred["start_logit"]
+                - best_non_null_pred["end_logit"]
             )
+            scores_diff_json[example.qas_id] = float(
+                score_diff
+            )  # To be JSON-serializable.
+            if score_diff > null_score_diff_threshold:
+                all_predictions[example.qas_id] = ""
+            else:
+                all_predictions[example.qas_id] = best_non_null_pred["text"]
+
+        # Make `predictions` JSON-serializable by casting numpy.float back to float.
+        all_nbest_json[example.qas_id] = [
+            {
+                k: (
+                    float(v)
+                    if isinstance(v, (numpy.float16, numpy.float32, numpy.float64))
+                    else v
+                )
+                for k, v in pred.items()
+            }
+            for pred in predictions
+        ]
+
+        if self.output_dir is not None:
```

Comment:
let's move this section to a helper function to maintain the flow in postprocessing